DataFormats/EcalDigi/src/EcalEBClusterTriggerPrimitiveDigi.cc

```diff
@@ -0,0 +1,88 @@
+#include "DataFormats/EcalDigi/interface/EcalEBClusterTriggerPrimitiveDigi.h"
+#include <iostream>
+
+
+EcalEBClusterTriggerPrimitiveDigi::EcalEBClusterTriggerPrimitiveDigi() : size_(0), data_(MAXSAMPLES) {
+}
+//EcalTriggerPrimitiveDigi::EcalTriggerPrimitiveDigi(const EcalTrigTowerDetId& id) : id_(id),
+//size_(0), data_(MAXSAMPLES) {
+//}
+
+EcalEBClusterTriggerPrimitiveDigi::EcalEBClusterTriggerPrimitiveDigi(const EBDetId& tpid, const std::vector<EBDetId>& xtalIds, float etaClu, float phiClu) : 
+  tpId_(tpid), 
+  cryIdInCluster_(xtalIds),  
+  etaClu_(etaClu),
+  phiClu_(phiClu),
+  size_(0), data_(MAXSAMPLES) {
+}
+
+void EcalEBClusterTriggerPrimitiveDigi::setSample(int i, const EcalEBClusterTriggerPrimitiveSample& sam) 
+{
+//  std::cout << " In setSample  i " << i << "  sam " << sam << std::endl;  
+  data_[i]=sam;
+//  std::cout << " In setSample data_[i] " << data_[i] << std::endl;  
+  
+}
+
+int EcalEBClusterTriggerPrimitiveDigi::sampleOfInterest() const
+{
+  if (size_ == 1)
+    return 0;
+  else if (size_ == 5)
+    return 2;
+  else
+    return -1;
+} 
+
+/// get the encoded/compressed Et of interesting sample
+int EcalEBClusterTriggerPrimitiveDigi::encodedEt() const 
+{
+  int sample = sampleOfInterest();
+  if (sample != -1)
+    return data_[sample].encodedEt();
+  else
+    return -1;
+}
+ 
+
+bool EcalEBClusterTriggerPrimitiveDigi::l1aSpike() const
+{
+  int sample = sampleOfInterest();
+  if (sample != -1)
+    return data_[sample].l1aSpike();
+  else
+    return -1;
```

Comment:
I encoded these lines *exaclty* as done here https://github.com/cms-sw/cmssw/blob/master/DataFormats/EcalDigi/src/EcalTriggerPrimitiveDigi.cc#L52 (except the name of the function). Which is not a bool, it's an int. And as far as I get it, the -1 is a clear wrong return value which can be filtered downstream. So I am not going to touch it

Reply:
```EcalEBClusterTriggerPrimitiveDigi::l1aSpike()``` returns a bool
If sample is "-1" it returns "true": if this is what you want, please write explicitely "true" instead of "-1"; if you want something else, please code for it.

If the code from which you derived this piece is also wrong, please also fix it (but the code you are pointing out returns an ```int```, not a ```bool```, and therefore I don't see issues with it)